balena/models/device.py

```diff
@@ -709,18 +717,46 @@ def move(
 
         self.__set(uuid_or_id, {"belongs_to__application": app["id"]})
 
-    def ping(self, uuid_or_id: Union[str, int]):
+    def __supervisor_request(self, method: str, path: str, body: Optional[AnyObject] = None):
+        params = {"apikey": self.__supervisor_api_key}
+        req = with_supervisor_locked_error(
+            lambda: requests.request(
+                method=method, url=urljoin(self.__supervisor_address, path), json=body, params=params  # type: ignore
+            )
+        )
+
+        if req.ok:
+            try:
+                return req.json()
+            except Exception:
+                return req.content.decode()
+        else:
+            raise exceptions.RequestError(body=req.content.decode(), status_code=req.status_code)
+
+    def __should_run_on_device(self, uuid_or_id: Optional[Union[str, int]]) -> bool:
+        return self.__on_device and (uuid_or_id is None or uuid_or_id == self.__local_device_uuid)
+
+    def ping(self, uuid_or_id: Optional[Union[str, int]] = None) -> None:
         """
         Ping a device.
         This is useful to signal that the supervisor is alive and responding.
 
         Args:
-            uuid_or_id (Union[str, int]): device uuid (str) or id (int).
+            uuid_or_id (Optional[Union[str, int]]): device uuid (str) or id (int) or None for sdk running on device.
```

Comment:
@pranasziaukas I prefer to leave "None" specifically because this is the actual check we do later (uuid_or_id is None). If you don't pass anything (none) it happens that default parameter is "None", but I would prefer to check for the actual Python keyword